diffkemp/semdiff/function_diff.py

```diff
@@ -227,30 +242,14 @@ def functions_diff(mod_first, mod_second,
                                            [Result.Kind.UNKNOWN,
                                             Result.Kind.ASSUMED_EQUAL]])
 
-                if prev_result_graph:
+                if prev_result_graph and not simplify:
```

Comment:
This works, but I think there might be an even better solution. When `simplify` is `True`, it means that SimpLL will be re-run anyway, therefore it is unnecessary to do the function cache update and `graph_to_fun_pair_list`. This would also allow us to get rid of the function cache rollback.

Reply:
I moved the cache update into this condition. May I remove the cache rollback completely?

Reply:
Yes, now the cache is not updated when `simplify` is `True`, therefore there is nothing to roll back.

Also the call to `graph_to_fun_pair_list` can be moved outside the `while simplify` loop.

Reply:
Done, please review.